app/code/Magento/Catalog/Model/Category/Link/SaveHandler.php

```diff
@@ -119,9 +119,11 @@ private function mergeCategoryLinks($newCategoryPositions, $oldCategoryPositions
 
             if ($key === false) {
                 $result[] = $newCategoryPosition;
-            } elseif ($oldCategoryPositions[$key]['position'] != $newCategoryPosition['position']) {
-                $result[] = $newCategoryPositions[$key];
-                unset($oldCategoryPositions[$key]);
+            } elseif (isset($oldCategoryPositions[$key])) {
```

Comment:
Personally I do not think we need the nested if here. I think we could cover this check via

```
} elseif (
    isset($oldCategoryPositions[$key])
    && $oldCategoryPositions[$key]['position'] != $newCategoryPosition['position']
) {
```

Or have I missed something?

Reply:
Yes, you are right. It is fixed and squashed.